core/capabilities/remote/trigger_publisher.go

```diff
@@ -162,16 +163,17 @@ func (p *triggerPublisher) Receive(_ context.Context, msg *types.MessageBody) {
 		}
 		ctx, cancel := p.stopCh.NewCtx()
 		callbackCh, err := p.underlying.RegisterTrigger(ctx, unmarshaled)
-		cancel()
 		if err == nil {
```

Comment:
nit/ Why do we handle the success case first? Same with the surrounding block. Usually the linter suggests error handling first, with a return, so that the subsequent logic can be inline instead of an indented block. Typically makes things much easier to read.